src/wp-includes/class-wp-theme.php

```diff
@@ -1998,16 +2002,43 @@ private function set_pattern_cache( array $patterns ) {
 			'version'  => $this->get( 'Version' ),
 			'patterns' => $patterns,
 		);
-		wp_cache_set( 'wp_theme_patterns_' . $this->stylesheet, $pattern_data, 'theme_files' );
+
+		/**
+		 * Filters the cache expiration time for theme files.
+		 *
+		 * @since 6.6.0
+		 *
+		 * @param int    $cache_expiration Cache expiration time in seconds.
+		 * @param string $cache_type       Type of cache being set.
+		 */
+		$cache_expiration = (int) apply_filters( 'wp_theme_files_cache_ttl', self::$cache_expiration, 'theme_block_patterns' );
+
+		// We don't want to cache patterns infinitely.
+		if ( $cache_expiration <= 0 ) {
```

Comment:
I don't think this is a blocker, but I don't think we should disallow setting these to never expire. I think a better validation would be to not cast the return value above and throw a doing it wrong if the return value is not an `int`. That would keep someone returning `false` thinking they were turning off caching from accidentally caching this indefinitely. Happy to go with this for now and consider as a follow-up.

Reply:
@joemcgill That would work for me, though my follow up question would be how you would "sanitize" the `false` then? Simply use the original `self::$cache_expiration`?

Reply:
Yep, we'd basically do the same thing we're already doing, but instead of `if ( $cache_expiration <= 0 ) { ...` we'd do `if ( ! is_int( $cache_expiration ) { ...`.

Reply:
@kt-12 had originally implemented it this way actually and I had suggested to be more "forgiving". Your argument around `false` makes sense to me, but my concern was more about someone returning `'30'` and that value failing, which really it doesn't have to fail since a string containing a number could be perfectly fine to let through (and cast to an integer).

But I'm onboard with your suggestion, and even with my caveat, there's always a benefit to use strict typing anyway.